src/or_trajopt/planning.py

```diff
@@ -632,3 +647,45 @@ def _WaypointsToTraj(self, robot, waypoints):
         for (i, waypoint) in enumerate(waypoints):
             traj.Insert(i, waypoint)
         return traj
+
+    def _raiseCollisionErrorForPose(self, robot, pose):
+        """ Identify collision for pose and raise error.
+        This will raise error even if no specific collision is found, 
+        so it should be called only when there is no IK solution. 
+        """
+        from openravepy import (IkFilterOptions,
+                                IkParameterization,
+                                IkParameterizationType)
+
+        manipulator = robot.GetActiveManipulator()
+
+        ik_param = IkParameterization(
+            pose, IkParameterizationType.Transform6D)
+
+        ik_solutions = manipulator.FindIKSolutions(
+            ik_param,
+            openravepy.IkFilterOptions.IgnoreSelfCollisions,
+            ikreturn=False,
+            releasegil=True
+            )
+
+        p = openravepy.KinBody.SaveParameters
+        with robot.CreateRobotStateSaver(p.ActiveDOF):
```

Comment:
Also, I would prefer that you move this inside `_checkCollisionForIKSolutions`, where the `SetActiveDOFValues` call actually happens.